### DISCUSSION TITLE
Add support for query builder

### DISCUSSION THREAD

**Discussion by imfaisii** [Power-Components/livewire-powergrid#976] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request
Curently:

public function datasource(): Collection
 {
     return User::table($this->tableName)->get();
 }
## Summary

Currently having problems to show the table data where there is no model present. Getting all data in a collection does not seem to be a good way of handling large data sets.

### Why is this needed?

This is will save memory for laravel application where table data is in millions.

### Suggested Solution(s)

public function datasource(): Collection
{
   return DB::table($this->tableName);
}

### Related issue(s)/PR(s)

Let us know if this is related to any issue/request

**Comment by luanfreitasdev** [comment]
Thanks a lot for your suggestion. I've been working on it and now it's available.

https://github.com/Power-Components/livewire-powergrid/releases/tag/v4.5.0
